### PR TITLE
add approximation for gelu, test=develop

### PR DESCRIPTION
**Pull request by iclementine**
1. add approximation for gelu, default value is False for backward compatability.
2. minor fix for erf_op: make Apply() of GradMaker protected method.

### CODE REVIEW

**Comment by iclementine** [comment]
文档修改在这里。https://github.com/PaddlePaddle/FluidDoc/pull/1904

Screenshot_2020-03-12 gelu — fluiddoc 1 7 文档

File: paddle/fluid/operators/gelu_op.cc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+};
+
+class GeluOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "Input of Gelu operator");
+    AddOutput("Out", "Output of Gelu operator");
+    AddAttr<bool>("approximate",
+                  "(bool, default false) use approximation of gelu")
+        .SetDefault(false);
```
Comment (lanxianghit) [inline paddle/fluid/operators/gelu_op.cc:88]: 默认值为false有特殊考虑吗，还是之前有很多模型使用了非近似计算方式？

Comment (iclementine) [reply paddle/fluid/operators/gelu_op.cc:88]: 之前的模型使用了非近似的计算方式。
模型库里涉及到的有
https://github.com/PaddlePaddle/models/search?q=gelu&unscoped_q=gelu

ERNIE 中有
https://github.com/PaddlePaddle/ERNIE/search?q=gelu&unscoped_q=gelu

**Review by lanxianghit — APPROVED** [review]

**Review by Boyan-Liu — APPROVED** [review]

**Comment by ddokupil** [comment]
Hi,
We noticed a 18% drop of BERT inference performance after this merge on CLX processor.
To reproduce drop you may use:
/repos/paddle_paddle/build/paddle/fluid/inference/tests/api/test_analyzer_bert --gtest_filter=Analyzer_bert.profile --paddle_num_threads=1 --repeat=1000 -use_mkldnn=true --infer_data=/repos/paddle_paddle/build/third_party/inference_demo/bert_emb128/data.txt --batch_size=1 --infer_model=/repos/paddle_paddle/build/third_party/inference_demo/bert_emb128/model

File: paddle/fluid/operators/gelu_op.cc [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                  "(bool, default false) Only used in mkldnn kernel")
+        .SetDefault(false);
+    AddAttr<bool>("use_cudnn",
+                  "(bool, default false) Only used in cudnn kernel, need "
+                  "install cudnn")
+        .SetDefault(false);
+    AddAttr<bool>("is_test",
+                  "(bool, default false) Set to true for inference only, false "
+                  "for training. Some layers may run faster when this is true.")
+        .SetDefault(false);
```
Comment (luotao1) [inline paddle/fluid/operators/gelu_op.cc:99]: `is_test` is not used in gelu_op, and #23318 will delete it at first.

**Pull request merged**